### DISCUSSION TITLE
[Metrics] Old metrics can also be got by port 15020 after customizing metrics.

### DISCUSSION THREAD

**Discussion by xjtupdy** [istio/istio#49626] [Q&A]
I use EnvoyFilter or Telemetry API to remove some labels of Istio metrics.
After deploying, new metrics with less labels can be generated. But Old metrics with all standard labels can also be got by ip:15020/stats/prometheus although they don't change any more.
Is it possible to remove these old metrics except redeploying pods?

**Comment by zirain** [comment]
try https://github.com/istio/istio/blob/faf5bdd3d5ff5fa2a7a8839f5b9980c84c6b4c6e/pilot/pkg/features/pilot.go#L207 with latest version
